pypeit/core/gui/skysub_regions.py

```diff
@@ -0,0 +1,657 @@
+"""
+This script allows the user to manually select the sky background regions
+"""
+
+import os, sys
+import copy
+import numpy as np
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib.widgets import Button, Slider
+import matplotlib.transforms as mtransforms
+from scipy.interpolate import RectBivariateSpline
+
+from pypeit import specobjs
+from pypeit import msgs
+from pypeit.io import write_to_fits
+from pypeit.core import pixels
+
+operations = dict({'cursor': "Select object trace (LMB click)\n" +
+                   "         Navigate (LMB drag = pan, RMB drag = zoom)\n" +
+                   "         Note: In order to pan/zoom, you need to first activate\n" +
+                   "         the pan/zoom tool with the 'p' key, or by selecting the\n" +
+                   "         pan/zoom tool on the Matplotlib navigation tool menu. You\n" +
+                   "         can also zoom using the magnifying glass (select this option\n" +
+                   "         from the Matplotlib navigation tool menu)\n",
+                   'a': "Add a new object trace using the selected method",
+                   'c': "Center the window at the location of the mouse",
+                   'd': "Delete selected object trace",
+                   'h/r': "Return zoom to the original plotting limits",
+                   'p': "Toggle pan/zoom with the cursor",
+                   '?': "Display the available options",
+                   })
+
+
+class SkySubGUI(object):
+    """
+    GUI to interactively define the sky regions. The GUI can be run within
+    PypeIt during data reduction, or as a standalone script outside of
+    PypeIt. To initialize the GUI, call the initialize() function in this
+    file.
+    """
+
+    def __init__(self, canvas, image, frame, outname, det, tslits_dict, axes,
+                 printout=False, runtime=False, resolution=1000):
+        """Controls for the interactive sky regions definition tasks in PypeIt.
+
+        The main goal of this routine is to interactively select sky background
+        regions.
+
+        Args:
+            canvas : Matploltib figure canvas
```

Comment:
These docs look like a mix of Numpy and Google.
Please go with one or the other.

Reply:
Right - I *think* I've cleared up the docs...